agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -1,173 +1,86 @@
- * throw an exception, or simply print a message and continue.
+ * The {@code CallReplacementTransformer} replaces each call to method m1 by a call to method m2. It
+ * is used by the MapCallsAgent.
+ *
+ * @see MapCallsAgent
  */
 public class CallReplacementTransformer implements ClassFileTransformer {
 
-  boolean debug = false;
-  boolean log_on = false;
-  boolean debug_class = false;
-
-  /** current Constant Pool */
-  static ConstantPoolGen pgen = null;
-
-  protected InstructionFactory ifact;
-
   /** Debug information about which classes are transformed and why */
-  public static SimpleLog debug_transform = new SimpleLog(false);
+  private static SimpleLog debug_transform =
+      new SimpleLog(
+          MapCallsAgent.debugPath + File.separator + "transform-log.txt", MapCallsAgent.debug);
 
-  public static SimpleLog debug_instrument = new SimpleLog(false);
-  public static SimpleLog debug_instrument_inst = new SimpleLog(false);
+  private static SimpleLog debug_instrument_inst =
+      new SimpleLog(
+          MapCallsAgent.debugPath + File.separator + "instrument-log.txt", MapCallsAgent.debug);
 
   /** Debug information on method maping */
-  public static SimpleLog debug_map = new SimpleLog("method_mapping.txt", true);
-
-  /** Class that defines a method (by its fully-qualified name and argument types) */
-  static class MethodDef {
-    /** Fully-qualified method name, such as "javax.swing.JOptionPane.showMessageDialog" */
-    String name;
-    /** The argument types */
-    Type[] arg_types;
-
-    MethodDef(String name, Type[] arg_types) {
-      this.name = name;
-      this.arg_types = arg_types;
-    }
-
-    boolean equals(String name, Type[] arg_types) {
-      if (!name.equals(this.name)) {
-        return false;
-      }
-      if (this.arg_types.length != arg_types.length) {
-        return false;
-      }
-      for (int ii = 0; ii < arg_types.length; ii++) {
-        if (!arg_types[ii].equals(this.arg_types[ii])) {
-          return (false);
-        }
-      }
-      return (true);
-    }
+  private static SimpleLog debug_map =
+      new SimpleLog(
+          MapCallsAgent.debugPath + File.separator + "method_mapping.txt", MapCallsAgent.debug);
 
-    @Override
-    public boolean equals(Object obj) {
-      if (!(obj instanceof MethodDef)) {
-        return false;
-      }
-      MethodDef md = (MethodDef) obj;
-      return equals(md.name, md.arg_types);
-    }
-
-    @Override
-    public int hashCode() {
-      int code = name.hashCode();
-      for (Type arg : arg_types) code += arg.hashCode();
-      return code;
-    }
+  /** Map from a method to its replacement. */
+  private final ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap;
 
-    @Override
-    public String toString() {
-      return String.format("%s (%s)", name, UtilMDE.join(arg_types, ", "));
-    }
-  }
+  /** The list of package prefixes (package name + ".") to exclude from transformation. */
```

Comment:
The constructor argument list, and uses in the code, have these fields in a different order than the order in which they are declared.  Please make these consistent, which make the code easier to read.